appdatabase/migrations/sql/1691258252_add_scope_to_pending.up.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TABLE pending_tx_status (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+INSERT INTO pending_tx_status (id, name) VALUES
+(0, 'Pending'),
+(1, 'Confirmed'),
+(2, 'Failed');
+
+ALTER TABLE pending_transactions ADD status INTEGER NOT NULL DEFAULT 0;
+
+-- The watcher will auto delete the pending txs that are confirmed or failed
+-- Else the producer module will have to delete them manually on processing
+ALTER TABLE pending_transactions ADD COLUMN auto_delete BOOLEAN NOT NULL DEFAULT 1;
```

Comment:
hmm how would we handle pending txs with `auto_delete` set to 0 if the app is closed and re-opened? Would the producer be in charge of loading any pending transactions of a particular type and listening for state changes? Do we really need this or can we leave processing of these entries fully in charge of the new module?